### PR TITLE
[webui] Introduce BranchPackage model test

### PR DESCRIPTION
**Pull request by ChrisBr**

### CODE REVIEW

File: src/api/spec/models/branch_package_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe BranchPackage, vcr: true do
+  let(:user) { create(:confirmed_user, login: 'tom') }
+  let(:home_project) { user.home_project }
+  let!(:project) { create(:project, name: 'BaseDistro') }
+  let!(:package) { create(:package, name: 'test_package', project: project) }
+
+  context '#branch' do
+    let(:branch) { BranchPackage.new(project: project.name, package: package.name).branch }
```
Comment (bgeuken) [inline src/api/spec/models/branch_package_spec.rb]: We should only put variables into let and let! Use shared context or simple methods for other stuff.

Comment (bgeuken) [inline src/api/spec/models/branch_package_spec.rb]: Also we should use factory girl here.

File: src/api/spec/models/branch_package_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    let!(:update_project) { create(:project, name: 'BaseDistro:Update') }
+    let(:update_attrib_type) { AttribType.find_by_namespace_and_name!('OBS', 'UpdateProject') }
+    let(:attrib_value) { AttribValue.new(value: 'BaseDistro:Update') }
```
Comment (bgeuken) [inline src/api/spec/models/branch_package_spec.rb]: and here

File: src/api/spec/models/branch_package_spec.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    end
+
+    context 'package with UpdateProject attribute' do
+      it 'should increase Package by one' do
+        expect { branch }.to change{ Package.count }.by(1)
+      end
+
+      it 'should create home:tom:branches:BaseDistro:Update project' do
+        branch
+        expect(Project.where(name: "#{home_project.name}:branches:BaseDistro:Update").exists?).to be true
```
Comment (bgeuken) [inline src/api/spec/models/branch_package_spec.rb]: ```expect(foo).to exist``` ;-)

Comment (ChrisBr) [inline src/api/spec/models/branch_package_spec.rb]: Ah, cool! Thx, will change that!

File: src/api/spec/models/branch_package_spec.rb (hunk 1 above)
Comment (ChrisBr) [inline src/api/spec/models/branch_package_spec.rb]: BranchPackage is a helper, so not sure if we should introduce a factory for it atm

``
src/api/app/helpers/branch_package.rb
``

Comment (ChrisBr) [inline src/api/spec/models/branch_package_spec.rb]: But I will move the function call to the tests!

Comment (ChrisBr) [inline src/api/spec/models/branch_package_spec.rb]: Thinking about the helper, ``spec/models/`` is then also not the right place for the test?
@bgeuken what do you think?

Comment (bgeuken) [inline src/api/spec/models/branch_package_spec.rb]: Yep, afaik it should go into spec/helpers/webui/ then.

**Comment by ChrisBr** [comment]
@bgeuken fixed your suggestions.

Furthermore, I moved BranchPackage from helpers to models.
```
 src/api/app/helpers/branch_package.rb → src/api/app/models/branch_package.rb 
```

Point is, it's not a module and we don't even use it as a helper, so I think model folder is more appropriate. Even it's not possible to test it as a helper because you can only test modules in rspec helper tests.

@hennevogel ^^

**Comment by bgeuken** [comment]
@ChrisBr Sounds a bit like something for lib/ then. \*hides\*

**Comment by hennevogel** [comment]
I know this pattern as 'service module' and people usually put them into app/models. As we have a ton of other classes which are not ActiveRecord in there already I guess it's okay. [People who are really nifty split their app/models...](https://blog.engineyard.com/2014/keeping-your-rails-controllers-dry-with-services)

**Comment by ChrisBr** [comment]
So @bgeuken and @hennevogel what is the final conclusion now?

**Comment by bgeuken** [comment]
@ChrisBr I can live with having this in app/models/

**Comment by hennevogel** [comment]
Like I said, me too :-)

**Pull request merged**